include/libtorrent/alert_types.hpp

```diff
@@ -247,9 +250,19 @@ TORRENT_VERSION_NAMESPACE_2
 #endif
 
 	// The ``torrent_removed_alert`` is posted whenever a torrent is removed. Since
-	// the torrent handle in its base class will always be invalid (since the torrent
+	// the torrent handle in its base class will usually be invalid (since the torrent
 	// is already removed) it has the info hash as a member, to identify it.
 	// It's posted when the ``status_notification`` bit is set in the alert_mask.
+        //
+        // Note that the ``handle`` may stay valid for an arbitrary time after
+        // ``torrent_removed_alert`` is posted, as long as some internal
+        // libtorrent task (such as an I/O task) refers to it. Additionally,
+        // other alerts like ``save_resume_data_alert`` may be posted after
+        // ``torrent_removed_alert``. To synchronize on whether the torrent has
+        // been removed or not, use the expression
+        // ``session.find_torrent(handle.info_hash()) == handle``. This
```

Comment:
Would it be better if the `torrent_handle` was always invalid after the torrent has been removed?
It would be quite simple to make that happen. Then this "hack" wouldn't be necessary, but would anything break in that case?

Reply:
Interesting. Would the session ignore calls for invalidated torrents?

What happens in the case of
```python
session.remove_torrent(handle)
handle.save_resume_data()
```
Presumably `save_resume_data()` doesn't throw because they're both asynchronous, but then how does `save_resume_data[_failed]_alert` get generated?